add incremental landing page (#5099)
dbt-labs/docs.getdbt.com: website/docs/docs/build/incremental-overview.md

---
title: "About Incremental models"

> description: "Learn about incremental models, when to use them, and how they work in dbt."

Reviewer: Great idea breaking this page up into it's own section! For our search and Google results, it might be beneficial to leave the word "Models" in the title and ID.

The descriptions for this page and the existing are very similar. To further distinguish between the two, it could help to have the word "Introduction" in this title and "Configuring" in the older article. So it would be
Introduction to incremental models 
and
Configuring incremental models 

to really drive home the differences between these two pages when folks are looking.

Author: i did try to be more explicit but what i didn't like was that the 'configure' was adding a 2nd line so i opted out. but it might be best to make sure it's super clear

Author: ![Screenshot 2024-03-18 at 15 22 37](https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/619e0dc0-8d4b-4845-a687-98c2827b50cb)

id: "incremental-overview"
pagination_next: "docs/build/incremental-models"
pagination_prev: null
---

Incremental models in dbt is a [materialization](/docs/build/materializations) strategy designed to efficiently update your data warehouse tables by only transforming and loading new or changed data since the last run. Instead of processing your entire dataset every time, incremental models append or update only the new rows, significantly reducing the time and resources required for your data transformations.

> This overview page will provide you with a brief overview of incremental models, their importance in data transformations, and the core concepts of incremental materializations in dbt.

Reviewer: ```suggestion
This page will provide you with a brief overview of incremental models, their importance in data transformations, and the core concepts of incremental materializations in dbt.
```

> <Lightbox src="/img/docs/building-a-dbt-project/incremental-diagram.jpg" width="60%" title="A visual representation of how incremental models work. Source: Materialization best practices page." />

Reviewer: We can make the title link to the best practice page
```suggestion
<Lightbox src="/img/docs/building-a-dbt-project/incremental-diagram.jpg" width="60%" title=<a href="https://docs.getdbt.com/best-practices/materializations/1-guide-overview"> A visual representation of how incremental models work. Source: Materialization best practices guide.</a> />
```

## Understand incremental models

Incremental models enable you to significantly reduce the build time by just transforming new records. This is particularly useful for large datasets, where the cost of processing the entire dataset is high.

> Incremental models [require extra configuration](/docs/build/incremental-models) and are an advanced usage of dbt. You can use it when your dbt runs are becoming too slow.

Reviewer: Should we say "too slow or costly"?
```suggestion
Incremental models [require extra configuration](/docs/build/incremental-models) and are an advanced usage of dbt. We recommend using them when your dbt runs are becoming too slow.
```

Author: it can be both, i think! for example, if they're model runs are taking a long time to run, it's a signal that they should think about their materializations. for example, the below dashboard in explorer lets users know their longest model-running time. so creating an incremental materialization will really reduce the build time by just transforming new records. and then help lower compute costs too.

![Screenshot 2024-03-18 at 14 48 48](https://github.com/dbt-labs/docs.getdbt.com/assets/89008547/96bfab69-69db-4cab-ae70-de8d2313a92f)

### When to use an incremental model

Building models as tables in your data warehouse is often preferred for better query performance. However, using `table` materialization can be computationally intensive, especially when:

- Source data has millions or billions of rows.

> - Data transformations on the source data are computationally expensive (take a long time to execute) and complex, like using Regex or UDFs.

Reviewer: ```suggestion
- Data transformations on the source data are computationally expensive (take a long time to execute) and complex, like when using Regex or UDFs.
```
